From 525c5839928c11ea9597c6c3388ba0fcb539e479 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Tue, 20 Mar 2018 16:56:39 +0800
Subject: [PATCH] ARM64: dts: rockchip: remove hdmi-codec node on
 rk3399-sapphire-excavator-linux and rk3399-videostrong-linux

hdmi_sound is defined in "rk3399-sapphire.dtsi"
So we do not need to define it again.

Change-Id: Ieb7e11240d06d64c82954ee725589b667480a4cc
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 .../rk3399-sapphire-excavator-linux.dts       | 20 ++-----------------
 .../dts/rockchip/rk3399-videostrong-linux.dts | 20 ++-----------------
 2 files changed, 4 insertions(+), 36 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index dc549b2bf67d..10a7f4f7a96e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -76,19 +76,8 @@
 		};
 	};
 
-	hdmi_codec: hdmi-codec {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,name = "HDMI-CODEC";
-
-		simple-audio-card,cpu {
-			sound-dai = <&i2s2>;
-		};
-
-		simple-audio-card,codec {
-			sound-dai = <&hdmi>;
-		};
+	hdmi_sound: hdmi-sound {
+		status = "okay";
 	};
 
 	gpio-keys {
@@ -392,11 +381,6 @@
 	};
 };
 
-&i2s2 {
-	#sound-dai-cells = <0>;
-	status = "okay";
-};
-
 &pcie_phy {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
index 96e8d13bdd67..0d0bad8561cf 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
@@ -44,19 +44,8 @@
 		};
 	};
 
-	hdmi_codec: hdmi-codec {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,name = "HDMI-CODEC";
-
-		simple-audio-card,cpu {
-			sound-dai = <&i2s2>;
-		};
-
-		simple-audio-card,codec {
-			sound-dai = <&hdmi>;
-		};
+	hdmi_sound: hdmi-sound {
+		status = "okay";
 	};
 
 	rt5640-sound {
@@ -216,11 +205,6 @@
 	#sound-dai-cells = <0>;
 };
 
-&i2s2 {
-	#sound-dai-cells = <0>;
-	status = "okay";
-};
-
 &rkvdec {
 	status = "okay";
 	/* 0 means ion, 1 means drm */
-- 
2.35.3

